AI/Nullkiller/Behaviors/BuyArmyBehavior.cpp

```diff
@@ -28,29 +28,34 @@ Goals::TGoalVec BuyArmyBehavior::decompose(const Nullkiller * ai) const
 {
 	Goals::TGoalVec tasks;
 
-	if(ai->cb->getDate(Date::DAY) == 1)
-		return tasks;
-		
 	auto heroes = cb->getHeroesInfo();
 
 	if(heroes.empty())
 	{
 		return tasks;
 	}
 
+	ai->dangerHitMap->updateHitMap();
+
 	for(auto town : cb->getTownsInfo())
 	{
+		uint8_t closestThreat = UINT8_MAX;
```

Comment:
closestTreatTurn

Reply:
dangerHitMap also can give treat node for town->visitablePos() and it has fastestDanger. For towns only it also gives all possible treats. For all the rest tiles only fastest and strongest

Reply:
Should work, is not it?